app/Http/Controllers/Service/Admin/CentresController.php

```diff
@@ -99,4 +100,47 @@ public function store(AdminNewCentreRequest $request)
             ->route('admin.centres.index')
             ->with('message', 'Centre ' . $centre->name . ' created');
     }
+
+    /**
+     * Show the form for editing a Centre.
+     *
+     * @return Factory|View
+     */
+    public function edit($id)
+    {
+        $centre = Centre::where('id', $id)->first();
+        return view('service.centres.edit', compact('centre'));
+    }
+
+    /**
+     * Show the form for editing a Centre's name.
+     *
+     * @return Factory|View
+     */
+    public function update(AdminUpdateCentreRequest $request)
```

Comment:
Laravel magic can get the dependency injector to pull the id out of the route parameters IF we send id as part of the URL  by doing `update(AdminUpdateCentreRequest $request, integer $id)`

if you type hint  `update(AdminUpdateCentreRequest $request, Centre $id)` it will _also_ do an automagical fetch of the model from the database. _however_ in this case it looks like you want to do that by hand with findOrFail so we can manually do something with the error?

Reply:
It took me a while to work it out, but this is changed.